django part 1/1. understanding

```diff
@@ -0,0 +1,60 @@
+1.  it's where you will create the ruleset of the db (how the db will look like)
+    it's in charge on how the db will look like (rules, set-up)
+
+2.  the views.py is where you will put all the logic and the functions. it's in charge of all the requests.
+
+3.  urls.py is the place that will tell the computer where to go when a certain word is put into the search bar.
+    it's in charge of the stream of info, and to direct where each info is stored.
+
+4.  migrations is what connects the model file and the db. general communication between the code and the db.
+    to create a new migration you will type "python manage.py makemigrations" and then "python manage.py migrate" to migrate the new rullset.
+
+5.  the serializer is in charge of converting complex data types (django models) into python data types (dictionaries).
+
+6.  djangorest (DRF) is a toolkit for building web apis in django (restful apis).
+
+7.  regular serializers are used when you want to have more control, flexibility and customization over the serialization and validation of the data.
+    it's good for when you want to work with data that is not directly tied to teh django model.
+
+    modelserializer is used when you want to do a simple copy/read/update/delete to a django model instance.
+
+    in summary i'll use regular serializers when i want to work with non django data, and modelserializer when it's django'y.
+
+8.  one-to-one, many-to-one, many-to-many.
+    one-to-one: one record of a particular model is related to exactly one record of another model. it's used for security purposes.
+    many-to-one: one record from the first model is related to many records from the second model. but one record from the second model is related to only one record from the first model. (album/song) 
```

Comment:
Good to note that it's implemented with `ForeignKeys`
You can also use it to create oneToMany relationship.